google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -43,6 +43,10 @@
 _delay: int = 55 * 60
 _sql_api_version: str = "v1beta4"
 
+# Time in seconds to wait for the result of fetching instance metadata before
+# timing out.
+_instance_metadata_timeout = 30
```

Comment:
IMO, this should be a user configurable timeout that's passed in via the function's signature. 

Is there anything the the DB API spec that defines a timeout arg?

Reply:
There's nothing in the DB API spec so I think it's driver specific: https://www.python.org/dev/peps/pep-0249/